src/main.rs

```diff
@@ -45,40 +45,208 @@ struct File {
+    Ok(path_log)
+}
+
+fn generate_json(log: &str, files: Vec<File>) -> String {
+    let mut data = json::JsonValue::new_object();
+    data["log"] = log.into();
+    data["files"] = json::JsonValue::new_array();
+    for file in files {
+        let mut file_data = json::JsonValue::new_object();
+        file_data["fileName"] = file.file_name.into();
+        file_data["fileContent"] = file.file_content.into();
+        data["files"].push(file_data).unwrap();
+    }
+    data.dump()
+}
+
+fn file_arr_from_path(dir_path: String) -> Result<Vec<File>, anyhow::Error> {
+    let mut file_arr: Vec<File> = vec![];
+
+    let dir = match fs::read_dir(dir_path.clone() + "/NM-migrated/system-connections") {
+        Ok(dir) => dir,
+        Err(_e) => fs::read_dir(dir_path + "/NM-migrated")?,
+    };
+
+    for dir_entry in dir {
+        let path = dir_entry?.path();
+        let file_type = match path.extension() {
+            Some(file_type) => match file_type.to_str().unwrap() {
+                "xml" => FileType::Xml,
+                _ => FileType::Ifcfg,
+            },
+            None => {
+                return Err(anyhow::anyhow!("File extension was not recognized"));
+            }
+        };
+        let file_contents = std::fs::read(&path).unwrap();
+        file_arr.push(File {
+            file_content: String::from_utf8(file_contents).unwrap(),
+            file_name: path.file_name().unwrap().to_str().unwrap().to_owned(),
+            file_type,
+        });
+    }
+    Ok(file_arr)
+}
+
+async fn return_config_json(uuid: OriginalUri, State(shared_state): State<AppState>) -> Response {
+    let database = shared_state.database.lock().await;
+
+    let uuid_stripped_of_prefix = match uuid.to_string().strip_prefix("/json/") {
+        Some(uri) => uri.to_owned(),
+        None => uuid.to_string(),
+    };
+
+    let path_log: (String, String) =
+        read_from_db(uuid_stripped_of_prefix.clone(), &database).unwrap();
+
+    let json_string = generate_json(
+        &path_log.1,
+        match file_arr_from_path(path_log.0.clone()) {
+            Ok(file_arr) => file_arr,
+            Err(e) => {
+                eprint!("Could not retrieve files from path {}: {}", path_log.0, e);
+                return StatusCode::INTERNAL_SERVER_ERROR.into_response();
+            }
+        },
+    );
+
+    match delete_file_from_db(&path_log.0, &uuid_stripped_of_prefix, &database) {
+        Ok(()) => (),
+        Err(e) => eprint!("Error when removing directory {}: {}", path_log.0, e),
     };
 
     drop(database);
 
-    let file_contents = match get_file_contents(std::path::Path::new("/tmp/").join(file_path)) {
+    axum::response::Json(json_string).into_response()
+}
+
+fn return_as_tar(path: String) -> anyhow::Result<tempfile::NamedTempFile> {
+    let output_tmpfile: tempfile::NamedTempFile = tempfile::Builder::new()
+        .prefix("nm-migrated.")
+        .suffix(".tar")
+        .keep(true)
```

Comment:
I think the `keep(true)` here is unnecessary. This file AFAICT remains in scope until it gets deleted. I'll leave it up to you though if you still want to `.close()` it to check for errors or if you trust in the deletion process :slightly_smiling_face: